service/service.go

```diff
@@ -0,0 +1,58 @@
+package service
+
+import (
+	"encoding/json"
+	"github.com/Jeffail/gabs"
+	"github.com/blockassets/cgminer_client"
+	"io"
+	"io/ioutil"
+	"log"
+	"os/exec"
+	"time"
+)
+
+//TODO: refactor this file to separate out the service level commands into their own files
+type Commands interface {
+	CgmQuit() error
+	Reboot()
+	UpdatePools(poolsAsJson io.ReadCloser, configFilePath string) error
+}
+type Command struct {
+}
+
+func (*Command) CgmQuit() error {
+	clnt := cgminer_client.New("localhost", 4028, 5*time.Second)
+	return clnt.Quit()
+}
+
+func (*Command) Reboot() {
+	time.Sleep(5 * time.Second)
+	log.Printf("Reboot Requested")
+	exec.Command("/sbin/reboot", "-f").Run()
+}
+
+type PoolAddresses struct {
+	Pool1 string `json:"pool1"`
+	Pool2 string `json:"pool2"`
+	Pool3 string `json:"pool3"`
+}
+
+func (*Command) UpdatePools(poolsAsJson io.ReadCloser, configFilePath string) error {
+	var pools PoolAddresses
+	err := json.NewDecoder(poolsAsJson).Decode(&pools)
+	if err == nil {
+		// use the gabs json helper..
+		cont := gabs.New()
+		cont, err = gabs.ParseJSONFile(configFilePath)
```

Comment:
I really don't think we need another dependency here. We should be able to use the standard library for this. Unmarshall into the struct. Set the fields on the struct and then Marshall the struct into a string.

You are missing the dependency in the Gopkg.toml file, so this would fail the build anyway.

Reply:
Ah - I ran deps ensure update.... not sure why it didnt get committed

Reply:
Definately understand your point. gabs seems to be used a lot and is a very slim helper on top of standard library. takes care of a lot of utitity it seems. But happy to change

Reply:
I'd like to suggest we consider using this library as one of our standards for managing json... As much as you dont like the (value, pathToValue), I like its use of 'path' to find values in the tree. Although this feature isn't needed in this context.

Ill remove it, but would like to keep discussion open for when you return

Reply:
Please read the docs on dep usage. When you add a dependency, you should dep ensure -add PATH first. You'd also see a diff for the Gopkg.toml file, so that is something you should look for too. =)

I'd prefer the config file have a struct around it for static typing and nice compile errors and not use a generic system. I don't think that library is a good idea unless you don't know the format of the json you are receiving, which is also a bad idea, you should know that. If we actually have a usecase of needing to find something with a path, then we can consider using it, but right now we really don't have that usecase. Try not to over engineer things. Simple first is always better.